src/logic/wallets/ethTransactions.ts

```diff
@@ -38,19 +39,13 @@ export const calculateGasPrice = async (): Promise<string> => {
   return Promise.reject(err)
 }
 
-export const calculateGasOf = async (txConfig: {
-  to: string
-  from: string
-  data: string
-  gasPrice?: number
-  gas?: number
-}): Promise<number> => {
-  const web3 = getWeb3()
+export const calculateGasOf = async (txConfig: EthAdapterTransaction): Promise<number> => {
   try {
-    const gas = await web3.eth.estimateGas(txConfig)
+    const ethAdapter = getSDKWeb3Adapter(txConfig.from)
 
-    return gas
-  } catch (err) {
+    return await ethAdapter.estimateGas(txConfig)
+  } catch {
+    const err = new CodedException(Errors._611, 'malformed transaction details')
     return Promise.reject(err)
   }
```

Comment:
```suggestion
  } catch (err) {
    throw new CodedException(Errors._611, err.message)
  }
```
It can fail for a number of reasons, I can imagine.